cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -415,8 +495,12 @@ struct DispatchBatchMemcpy : SelectedPolicy
     // The number of thread blocks (or tiles) required to process all of the given buffers
     BlockOffsetT num_tiles = DivideAndRoundUp(num_buffers, TILE_SIZE);
 
-    using BlevBufferSrcsOutItT        = void **;
-    using BlevBufferDstsOutItT        = void **;
+    using BlevBufferSrcsOutT =
+      typename ::cuda::std::conditional<memcpy, void *, cub::detail::value_t<InputBufferIt>>::type;
+    using BlevBufferDstOutT =
+      typename ::cuda::std::conditional<memcpy, void *, cub::detail::value_t<OutputBufferIt>>::type;
```

Comment:
Here, I changed `void *` to `uint8_t *` because a template check was giving a compile error on using void as a type in another templated type resolution conditional expression even though was `memcpy=true` and the other branch was not taken.

Reply:
I think we should fix the root cause of that issue. I left a suggestion [here](https://github.com/NVIDIA/cub/pull/675#discussion_r1170015035). Same goes for the type alias in the `agent`, as otherwise it could fail to compile if users provided pointer-to-void for a buffer's source.